ovh/resource_ovh_dns_record.go

```diff
@@ -0,0 +1,183 @@
+package ovh
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+type NewRecord struct {
+	Target    string `json:"target"`
+	Ttl       int    `json:"ttl"`
+	FieldType string `json:"fieldType"`
+	SubDomain string `json:"subDomain"`
+}
+
+type Record struct {
+	Id        int    `json:"id"`
+	Zone      string `json:"zone"`
+	Target    string `json:"target"`
+	Ttl       int    `json:"ttl"`
+	FieldType string `json:"fieldType"`
+	SubDomain string `json:"subDomain"`
+}
+
+func resourceOVHDomainZoneRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOVHRecordCreate,
+		Read:   resourceOVHRecordRead,
+		Update: resourceOVHRecordUpdate,
+		Delete: resourceOVHRecordDelete,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"target": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ttl": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "3600",
+			},
+			"fieldType": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"subDomain": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceOVHRecordCreate(d *schema.ResourceData, meta interface{}) error {
+	provider := meta.(*Config)
+
+	// Create the new record
+	newRecord := &NewRecord{
+		FieldType: d.Get("fieldType").(string),
+		SubDomain: d.Get("subDomain").(string),
+		Target:    d.Get("target").(string),
+	}
+
+	newRecord.Ttl, _ = strconv.Atoi(d.Get("ttl").(string))
+
+	log.Printf("[DEBUG] OVH Record create configuration: %#v", newRecord)
+
+	resultRecord := Record{}
+
+	err := provider.OVHClient.Post(
+		fmt.Sprintf("/domain/zone/%s/record", d.Get("zone").(string)),
+		newRecord,
+		&resultRecord,
+	)
+
+	if err != nil {
+		return fmt.Errorf("Failed to create OVH Record: %s", err)
+	}
+
+	d.SetId(strconv.Itoa(resultRecord.Id))
+	d.Set("id", strconv.Itoa(resultRecord.Id))
```

Comment:
@grubernaut as i understand terraform generate internally an ID for every ressource, 
the OVH api return a ID for each DNS entry so I force the internal terraform ID to be the ID return by the API
as this id is also part of the resource i'm setting it in the resource data schema

I don't know if i'm clear here and if this right way to do it